src/metabase/util.clj

```diff
@@ -786,3 +748,57 @@
   `Locale/US` locale."
   [^CharSequence s]
   (.. s toString (toLowerCase (Locale/US))))
+
+(defn format-nanoseconds
+  "Format a time interval in nanoseconds to something more readable (µs/ms/etc.)
+   Useful for logging elapsed time when using `(System/nanotime)`"
+  ^String [nanoseconds]
+  (loop [n nanoseconds, [[unit divisor] & more] [[:ns 1000] [:µs 1000] [:ms 1000] [:s 60] [:mins 60] [:hours 24]
```

Comment:
A comment here that [:ns 1000] is different as it's the init would be helpful as this is else quite confusing as it looks like conversion factors, but it's not throughout. Also, if you change weeks to infinity it will always be a sufficient stopping condition (but that might be too cute).

Reply:
that's a good point, I'll tweak that. I think infinity is a good idea, I was trying to achieve the same thing with `MAX_VALUE` anyway